bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -718,8 +718,14 @@ func (r *KubeadmConfigReconciler) storeBootstrapData(ctx context.Context, scope
 		Type: clusterv1.ClusterSecretType,
 	}
 
+	// as secret creation and scope.Config status patch are not atomic operations
+	// it is possible that secret creation happens but the config.Status patches are not applied
 	if err := r.Client.Create(ctx, secret); err != nil {
-		return errors.Wrapf(err, "failed to create bootstrap data secret for KubeadmConfig %s/%s", scope.Config.Namespace, scope.Config.Name)
+		if !apierrors.IsAlreadyExists(err) {
+			return errors.Wrapf(err, "failed to create bootstrap data secret for KubeadmConfig %s/%s", scope.Config.Namespace, scope.Config.Name)
+		} else {
+			r.Log.Info("bootstrap data secret for KubeadmConfig already exists", "secret", secret.Name, "KubeadmConfig", scope.Config.Name)
+		}
```

Comment:
We can probably drop the `else` block here and log regardless
```suggestion
	r.Log.Info("Bootstrap data secret for KubeadmConfig already exists", "secret", secret.Name, "KubeadmConfig", scope.Config.Name)
```

Reply:
removed else, as suggested